src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -355,6 +369,8 @@ public void StartProfileOptimization(string profile)
 #endif // FEATURE_MULTICOREJI
         }
         
+        public event ResolveEventHandler AssemblyResolve;
```

Comment:
Thread-safety: The assembly load context is safe to use from multiple threads. The compiler generated event glue is not thread safe. It needs to be done manually here - take a look at how it is done on AppDomain,